web-extension-1/src/popup/popup.tsx

```diff
@@ -1 +1,14 @@
-console.log('Popup script');
\ No newline at end of file
+// Popup script
+import { currentTabPromise, IPopupEvent } from '../interfaces/tab'
+
+(() => {
+  // fetch the current and active tab
+  // from which the popup as called
+  currentTabPromise().then(function(tabs) {
+    if (tabs[0] !== undefined) {
+      const tabId = tabs[0].id as number;
+      browser.runtime.sendMessage({ active : true, tabId: tabId } as IPopupEvent);
+    }
+  });
```

Comment:
You could have let the background process retrieve the current tab upon receiving the message to simplify this part